incubator/orm/index.go

```diff
@@ -23,15 +23,26 @@ type MultiKeyIndex struct {
 	prefix        byte
 	rowGetter     RowGetter
 	indexer       indexer
-	indexKeyCodec IndexKeyCodec
+	indexKeyCodec IndexKeyDecoder
 }
 
 // NewIndex builds a MultiKeyIndex
-func NewIndex(builder Indexable, prefix byte, indexer IndexerFunc) *MultiKeyIndex {
-	return newIndex(builder, prefix, NewIndexer(indexer, builder.IndexKeyCodec()))
+func NewIndex(builder Indexable, prefix byte, indexerFunc IndexerFunc) *MultiKeyIndex {
+	return newIndex(builder, prefix, NewIndexer(indexerFunc, builder.IndexKeyCodec()))
 }
 
-func newIndex(builder Indexable, prefix byte, indexer *Indexer) *MultiKeyIndex {
+// NewVirtualIndex provides an index API on top of a natural key table. The use case is
+// a combined natural key where the first element has a fixed size.
```

Comment:
What would you do if the first element doesn't have a fixed size? Is there a way you could use the max 255 dynamic length codec with this?

Reply:
Without a fixed size we would easily get overlapping results in a prefix scan. We may be able to filter them out with a custom iterator on "simple natural" keys but this is a very special use case.

The one I wanted to cover was a natural key starting with the groupID so that we can have a typed index:
`idx := AsUInt64Index(NewVirtualIndex(builder))` without extra costs.